src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -1280,6 +1286,43 @@ internal static int GetBinIdx(float output, float min, float binSize, int numBin
             return binIdx;
         }
 
+        bool ISingleCanSaveOnnx.SaveAsOnnx(OnnxContext ctx, string[] outputNames, string featureColumn)
+        {
+            _host.CheckValue(ctx, nameof(ctx));
+            _host.CheckValue(outputNames, nameof(outputNames));
+            _host.Check(Utils.Size(outputNames) == 2);
+
+            const int minimumOpSetVersion = 9;
+            ctx.CheckOpSetVersion(minimumOpSetVersion, "NaiveCalibrator");
+
+            string opType = "Sub";
+            var minVar = ctx.AddInitializer(Min, "Min");
+            var subNodeOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "subNodeOutput");
+            var node = ctx.CreateNode(opType, new[] { outputNames[0], minVar }, new[] { subNodeOutput }, ctx.GetNodeName(opType), "");
+
+            opType = "Div";
+            var binSizeVar = ctx.AddInitializer(BinSize, "BinSize");
+            var divNodeOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "binIndexOutput");
+            node = ctx.CreateNode(opType, new[] { subNodeOutput, binSizeVar }, new[] { divNodeOutput }, ctx.GetNodeName(opType), "");
+
+            opType = "Cast";
+            var castOutput = ctx.AddIntermediateVariable(NumberDataViewType.Int64, "castOutput");
+            node = ctx.CreateNode(opType, divNodeOutput, castOutput, ctx.GetNodeName(opType), "");
+            var toTypeInt = InternalDataKindExtensions.ToInternalDataKind(DataKind.Int64).ToType();
+            node.AddAttribute("to", toTypeInt);
+
```

Comment:
It turns out that this whole business of `InternalDataKindExtensions.ToInternalDataKind(DataKind.Int64).ToType()` just translates to `typeof(long)`. (See [KeyToVector.cs](https://github.com/dotnet/machinelearning/blob/bc10d609cc11dbf2c1aafaaf32ed25c6867e4a66/src/Microsoft.ML.Data/Transforms/KeyToVector.cs#L714)

In the next cleanup pass we can change the other places that use this pattern. But for now you can use `typeof(long)` directly.

Reply:
But, isn't it necessary to keep the whole `InternalDataKindExtensions.ToInternalDataKind(DataKinf.Int64).ToType()` call, in case that in the future, for whatever reason, we change the internal rawtype that represents `DataKind.Int64` and/or `InternalDataKind.I8`?

---
In reply to: [452946693](https://github.com/dotnet/machinelearning/pull/5289#discussion_r452946693) [](ancestors = 452946693)

Reply:
Ideally yes. But in this particular case, it is not a very convincing argument. We are already using two different types `NumberDataViewType.Int64` and `DataKind.Int64`. Also, after that whole translation, `toTypeInt` translates to `typeof(long)` and so we now have three different types which holds an underlying assumption that the three types are compatible. 

The right interface would actually have been to have to have `node.AddAttribute` directly take in either `NumberDataViewType.Int64` or `DataKind.Int64`.  As long as `node.AddAttribute` is expecting a native type, it is better to supply that native type directly. `node.AddAttribute("to", typeof(long))` is a lot more readable than what is above. 

---
In reply to: [452963359](https://github.com/dotnet/machinelearning/pull/5289#discussion_r452963359) [](ancestors = 452963359,452946693)

Reply:
Oh, I see. That makes sense. Thanks for the explanation.

---
In reply to: [452990378](https://github.com/dotnet/machinelearning/pull/5289#discussion_r452990378) [](ancestors = 452990378,452963359,452946693)